core/server/services/mega/mega.js

```diff
@@ -384,6 +385,11 @@ async function getEmailMemberRows({emailModel, memberSegment, options}) {
         filterOptions.filter = `${filterOptions.filter}+${memberSegment}`;
     }
 
+    if (labsService.isSet('multipleNewsletters') && emailModel.get('newsletter_id')) {
+        const newsletter = await emailModel.related('newsletter').fetch();
+        filterOptions.filter = `${filterOptions.filter}+newsletters:${newsletter.slug}`;
```

Comment:
While it's not used right now, I think we should also take into account the newsletter's `visibility` property here to satisfy [this requirement](https://www.notion.so/ghost/Architecture-Notes-65c8733988e946d98872bb9f4655b40f#e656ac2e3d974b2eaecc911812f0173a).